plot-doc/plot/scribblings/renderer2d.scrbl

```diff
@@ -890,3 +890,99 @@ Returns a renderer that draws a labeled point on a polar function's graph.
+
+@defmethod[(set-mouse-event-callback [callback (or/c plot-mouse-event-callback/c #f)]) any/c]{
+
+Set a callback function to be invoked with mouse events from the snip.  The
+callback is invoked with the actual snip object, the @racket[mouse-event%] and
+the X, Y position of the mouse in plot coordinates -- these are in the same
+coordinate system used by the renderers in the plot.  The X, Y values are
+@racket[#f] when the mouse is outside the plot area (for example, when the
+mouse is over the axis area).
+
+When a callback is installed, the default zoom functionality of the plot snips
+is disabled.  This can be restored by calling
+@racket[set-mouse-event-callback] with a @racket[#f] argument.
+
+}
+
+@defmethod[(set-overlay-renderers [renderers (or/c #f (treeof renderer2d?))]) any/c]{
+
+Set a list of renderers (or more generally, a tree of renderers) to be drawn
+on top of the existing plot.  This can be any combination of 2D renderers, but
+it will not be able to modify the axes or the dimensions of the plot area.
+Only one set of overlay renderers can be installed, calling this method a
+second time will replace the previous overlays.  Specifying @racket[#f] as the
+renderers will cause overlays to be disabled.
+
+}
+}
+
+@defthing[plot-mouse-event-callback/c contract? #:value (-> (is-a?/c snip%)
+                                                       (is-a?/c mouse-event%)
+                                                       (or/c real? #f)
+                                                       (or/c real? #f) any/c)]{
+A contract for callback functions passed to @racket[set-mouse-event-callback].
+}
+
+
+For example, if you evaluate the code below in DrRacket, the resulting plot
+will show a vertical line tracking the mouse and the current plot position is
+shown on a label.  This is achieved by adding a mouse callback to the plot
+snip returned by @racket[plot-snip].  When invoked, the mouse callback will
+add a @racket[vrule] at the current X position and a @racket[point-label] at
+the current value of the plotted function.
+
+@examples[#:eval plot-eval
+(require plot)
+(define snip (plot-snip (function sin) #:x-min -5 #:x-max 5))
+(define (mouse-callback snip event x y)
+   (if (and x y)
+       (send snip set-overlay-renderers (list (vrule x) (point-label (x (sin x)))))
+       (send snip set-overlay-renderers #f)))
+(send snip set-mouse-event-callback mouse-callback)
+snip]
+
+Here are a few hints for adding common interactive elements to racket plots:
+
+@itemlist[
+
+@item{@(racket hrule) and @(racket vrule) renderers can be used to draw
+horizontal and vertical lines that track the mouse position}
+
+@item{The @(racket rectangles) renderer can be used to highlight a region on
+the plot.  You can also specify @racket[-inf.0] and @racket[+inf.0] as limits
+for one of the dimensions, to extend the rectangle all the way to the edge of
+the plot in that direction.  You will need to specify a non zero value for the
```

Comment:
suggestion: delete these two sentences ("You can .... You will ....") and skip to the example